src/pwa-buildpack/project-setup/create-configuration-files/index.md

```diff
@@ -0,0 +1,268 @@
+
+* `MAGENTO_BACKEND_DOMAIN` - Your local Magento store’s host and port.
+* `MAGENTO_PATH` - The absolute path to the root directory of your local Magento store
+* `MAGENTO_BACKEND_PUBLIC_PATH` - Your theme’s vendor and name. The locale must be `en_US` for now.
+* `SERVICE_WORKER_FILE_NAME` - Set this value to `"sw.js"`.
+
+Your file should look like the following:
+
+``` text
+MAGENTO_BACKEND_DOMAIN=https://localhost.magento:8008
+
+MAGENTO_PATH=~/path/to/magento/rootdir
+
+MAGENTO_BACKEND_PUBLIC_PATH=/pub/static/frontend/OrangeCompany/orange-theme/en_US
+
+SERVICE_WORKER_FILE_NAME="sw.js"
+```
+
+## Create the webpack configuration file
+
+Create a `webpack.config.js` file in your theme’s root directory.
+This file exports a configuration object that tells Webpack how to build your theme.
+
+Webpack configuration is a complicated topic, so the following sections explain what each piece does in the configuration file.
+To see the fully assembled file, see the example [webpack.config.js].
+
+### Import environment variables
+
+At the top of the webpack.config.js file add the following:
+
+``` javascript
+require('dotenv').config();
+```
+
+This imports the the contents of the `.env` file as environment variables using the `dotenv` module.
+These environment variables are accessed using the `process.env` global object.
+
+For example, the following code outputs the value of the `MAGENTO_BACKEND_DOMAIN` environment variable:
+
+``` javascript
+console.log(process.env.MAGENTO_BACKEND_DOMAIN);
+```
+
+### Import Webpack and pwa-buildpack libraries
+
+Append the following content to `webpack.config.js` to import the Webpack and pwa-buildpack libraries:
+
+``` javascript
+const webpack = require('webpack');
+const {
+    Webpack: {
+        MagentoRootComponentsPlugin,
+        ServiceWorkerPlugin,
+        MagentoResolver,
+        PWADevServer
+    }
+} = require('@magento/pwa-buildpack');
+```
+
+### Define paths to theme resources
+
+Add the following content to `webpack.config.js` to define the paths to your theme resources:
+
+``` javascript 
+const path = require('path');
+
+const themePaths = {
+    src: path.resolve(__dirname, 'src'),
+    assets: path.resolve(__dirname, 'web'),
+    output: path.resolve(__dirname, 'web/js'),
+};
+```
+
+This snippet uses the `path` module to format and normalize the file paths.
+It also uses the special Node variable `__dirname` because it always resolves to the directory of the current executing script file.
+
+The locations specified in the snippet are the standard locations of source code, static assets, and build output in a [Peregrine] app.
+
+### Export the Webpack configuration object
+
+Append the following content to webpack.config.js to export the configuration object for Webpack:
+
+``` javascript
+module.exports = async function(env) {
+    const config = {
+        mode: env.mode, // passed on the command line via the '--env' flag
```

Comment:
This line that sets `mode` should be removed; that would be for Webpack 4, which we don't intend to support right away. It was my mistake originally.

**Also: All instances of `env.mode` must be replaced with `env.phase`.**